File: JAVA/451. Sort Characters By Frequency.java
class Solution {
    public String frequencySort(String s) {
        char[] chars = s.toCharArray(); 
        Map<Character, Integer> frequencyMap = new HashMap<>();
        for (char c : chars) {
            frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
        }
        List<Character> sortedChars = new ArrayList<>(frequencyMap.keySet());
        Collections.sort(sortedChars, (a, b) -> frequencyMap.get(b) - frequencyMap.get(a));
        StringBuilder result = new StringBuilder();
        for (char c : sortedChars) {
            int frequency = frequencyMap.get(c);
            for (int i = 0; i < frequency; i++) {
                result.append(c);
            }
        }
        return result.toString();
    }
}

File: C++/451. Sort Characters By Frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        vector<pair<int,char>>v;
        map<char,int>m;
        for(int i=0; i<s.size(); i++)m[s[i]]++;
        for(auto it:m)v.push_back({-1*it.second,it.first});
        sort(v.begin(),v.end());
        string res;
        for(auto it:v){
            for(int i=1; i<=abs(it.first); i++){
                res+=it.second;
            }
        }        
        return res;
    }
};
